core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -106,15 +86,16 @@ class SubmissionMonitorActor(val workspaceName: WorkspaceName,
       handleStatusResponses(response) pipeTo self
     case StatusCheckComplete(terminateActor) =>
       logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
-      if (terminateActor) stop(self)
+      // Before terminating this actor, run one more CheckCurrentWorkflowStatusCounts pass to ensure
+      // we have accurate metrics at the time of actor termination.
+      if (terminateActor) {
+        self ! CheckCurrentWorkflowStatusCounts
```

Comment:
but you're not waiting for this. does the message get processed if you immediately call `stop()` ? do you just want to call the function directly?